mreg_cli/api/models.py

```diff
@@ -532,20 +534,45 @@ def create_delegation(
         )
         return resp.ok if resp else False
 
-    def get_delegation(self, name: str) -> ForwardZoneDelegation | ReverseZoneDelegation:
+    def get_delegation(self, name: str) -> ForwardZoneDelegation | ReverseZoneDelegation | None:
         """Get a delegation by name.
 
-        :param zone: The zone to search in.
         :param name: The name of the delegation to get.
-        :returns: The delegation object.
+        :returns: The delegation object if found.
         """
+        self.ensure_delegation_in_zone(name)
         cls = Delegation.type_by_zone(self)
-
-        resp = get(cls.endpoint_with_id(self, name))
+        resp = get(cls.endpoint_with_id(self, name), ok404=True)
         if not resp:
-            raise EntityNotFound(f"Could not find delegation {name!r} in zone {name!r}")
+            return None
         return cls.model_validate_json(resp.text)
 
+    def get_delegation_or_raise(self, name: str) -> ForwardZoneDelegation | ReverseZoneDelegation:
```

Comment:
We need to add this as far as I understand because `Delegation.get_by_name_{and,or}_raise()` doesn't work when the endpoint has 2 parameters/path components it needs to replace. I.e.: `/api/v1/zones/forward/{}/delegations/{}`.

Reply:
Check point 2 of https://github.com/unioslo/mreg-cli/pull/227#issuecomment-2112481006 for more information.